src/util.test.js

```diff
@@ -0,0 +1,35 @@
+import { imageVariantURLStr } from './util';
+
+describe('util', () => {
+  describe('imageVariantURLStr', () => {
+    it('constructs a valid URL string', () => {
+      expect(
+        imageVariantURLStr({
+          w: 500,
+          h: 1000,
+          fit: 'scale',
+        })
+      ).toEqual('w:500;h:1000;fit:scale');
+    });
+
+    it('allows any number of options', () => {
+      expect(
+        imageVariantURLStr({
+          w: 500,
+          h: 2000,
+          fit: 'scale',
+          newparam: 'newvalue',
+        })
+      ).toEqual('w:500;h:2000;fit:scale;newparam:newvalue');
```

Comment:
Is there some param checks in API side? Our clients know that the API is currently on top of Imgix. They might try to test if imgix-specific params work => and if they do work, we are ~locked with Imgix.

Reply:
The API validates params and only allows the two fit modes, scale and crop, and even is a bit stricter than Imgix regarding w and h, limiting them between 16px and 3072px.

Reply:
@lyyder was there a performance/cost reason for 3072?
It's smaller than iMac 4K, 5K, Pro Display, etc.

Reply:
The lastest 16" MacBook was used as a reference. Also the ratio is not too large, so there's room to increase it if needed. The 4/5k point is valid, maybe the min size could be increased. After all, it does make sense to request images larger than are needed.

Reply:
I'll resolve this now. The max image size can be expanded in the API side if needed.